mixer/adapter/stackdriver/log/log.go

```diff
@@ -106,14 +118,36 @@ func (b *builder) Build(ctx context.Context, env adapter.Env) (adapter.Handler,
 			_ = logger.Errorf("failed to evaluate template for log %s, skipping: %v", name, err)
 			continue
 		}
+
+		if log.SinkInfo != nil {
+			// first try to CreateSink, if error is AlreadyExists then update that sink.
+			_, err = syncClient.CreateSink(context.Background(), &logadmin.Sink{
+				ID:          log.SinkInfo.Id,
+				Destination: log.SinkInfo.Destination,
+				Filter:      log.SinkInfo.Filter,
+			})
+			if err != nil {
+				if strings.Contains(err.Error(), "AlreadyExists") {
+					_, err = syncClient.UpdateSink(context.Background(), &logadmin.Sink{
+						ID:          log.SinkInfo.Id,
+						Destination: log.SinkInfo.Destination,
+						Filter:      log.SinkInfo.Filter,
+					})
+				}
+				if err != nil {
```

Comment:
nit: should this check be rolled inside the `if` above it (as that is the only way a subsequent check for `nil` is needed)?

maybe we could make this block clearer as follows:

```
if log.SinkInfo != nil {
  sink := &logadmin.Sink{...}
  ctx := context.Background()
  var sinkErr error
  if _, sinkErr = syncClient.CreateSink(ctx, sink); isAlreadyExists(sinkErr) {
     _, sinkErr = syncClient.UpdateSink(ctx, sink)
  }
  if sinkErr != nil {
     logger.Warningf(...)
  }
}

...

func isAlreadyExists(err error) bool {
   if err == nil {
      return false
   }
   return strings.Contains(err.Error(), "AlreadyExists")
}

Reply:
Thanks.. yes that looks much better.